proposal.rst

```diff
@@ -180,39 +180,57 @@ Motivations
+
+* Implement class for preparing data, i.e. download it and convert to suitable form. The most convenient dataset is `FIRE <http://fire.irsi.res.in/fire/static/data>`_ and `TREC <https://trec.nist.gov/data/t2002_filtering.html>`_. It is hard to fully generalize this case, so I prefer to implement parser for each dataset:
+
+	* FIRE queries stored in inappropriate format, they should be transformed to use them in Xapian; It will possible to use English language corpus, because Eglish stemmer exists in Xapian.
+
+	* TREC dataset has big enough amount of documents, it is possible to combine some of them during pasing to create more complex queries.
+
+* Implement evaluator. According to xapian-letor/tests/api_letor.cc it is no big difference in applying rankers to data, it should just apply algorithms and return scores, so it could be another one class in xapian/letor. It would be nice to be able to call it same way as make check.
+
+* Write suite for combining documents. TREC dataset has big corpus of small documents. To get more content small documents could be merged. Also queries and qrels have to be transformed. After that evaluation could be done using Xapian API logic operations for queries. Since it is not needed for every Letor run, it should be separate tool, like xapian-letor/bin/xapian-prepare-trainingfile.cc. 
+
+* It is possible to use Wikipedia dumps as data for perftest. `Wikipedia <https://dumps.wikimedia.org/enwiki/latest/>`_ corpus just stores XML file with links to articles and some info, so it each document should be downloaded and indexed individually. Also plain text `extractor <https://github.com/RaRe-Technologies/gensim/blob/develop/gensim/corpora/wikicorpus.py>`_ and query fabricator needed.
 
 **Do you have any preliminary findings or results which suggest that your
 approach is possible and likely to succeed?**
 
-I worked with Xapian test harness, it has BackendManager class wich makes testing databases api easier. I believe ranking benchmark should be done in similar way. 
+I worked with Xapian test harness in `#758 <https://trac.xapian.org/ticket/758>`_, it has BackendManager class wich makes testing databases api easier. I believe ranking benchmark should be done in similar way.
 
 **What other approaches have you considered, and why did you reject those in
 favour of your chosen approach?**
 
-Another way is to create ranking suite as part of Letor, but it is not clear enough how to develop performance measurement api in this case.
+I have discussed the possibility of using web datasets like INEX, but there are several reasons to store dataset locally:
+
+* Most of Web-API datasets requires an account creation, which is not suitable for Xapian users.
+
+* All data should be stored locally at least in parts to be indexed.
```

Comment:
Is this true? It might be more convenient, but "should" is a strong word. If you're indexing a million 1MB documents, that's a lot of storage that you'll need: for the source documents, then more for the database you're building. In that situation, would it be more useful to be able to index online documents instead?

(I'm *not* saying that you should take this route. I'm saying that you should be more precise in your statements in the proposal. If you say something like "we should do things this way" then you need to back that up with either references to existing work or research, or your own argument.)